src/array/dictionary/value_map.rs

```diff
@@ -15,10 +15,61 @@ use crate::{
 
 use super::DictionaryKey;
 
+/// Hasher for pre-hashed values; similar to `hash_hasher` but with native endianness.
+///
+/// We know that we'll only use it for `u64` values, so we can avoid endian conversion.
+///
+/// Invariant: hash of a u64 value is always equal to itself.
+#[derive(Copy, Clone, Default)]
+pub struct HashHasherNative(u64);
+
+impl Hasher for HashHasherNative {
+    #[inline]
+    fn write(&mut self, value: &[u8]) {
+        #[cfg(target_endian = "little")]
+        {
+            for (index, item) in value.iter().enumerate().take(8) {
```

Comment:
This is also sort of an overkill tbh... because we know that `value.len() == 8` always